pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -2069,6 +2121,23 @@ func (lbaas *LbaasV2) ensureLoadBalancer(ctx context.Context, clusterName string
 	return status, nil
 }
 
+func (lbaas *LbaasV2) listSubnetsForNetwork(networkID string) ([]subnets.Subnet, error) {
+	mc := metrics.NewMetricContext("subnet", "list")
+	allPages, err := subnets.List(lbaas.network, subnets.ListOpts{NetworkID: networkID}).AllPages()
+	if mc.ObserveRequest(err) != nil {
+		return nil, fmt.Errorf("error listing subnets of network %s: %v", networkID, err)
+	}
+	subs, err := subnets.ExtractSubnets(allPages)
+	if err != nil {
+		return nil, fmt.Errorf("error extracting subnets from pages: %v", err)
+	}
+
+	if len(subs) == 0 {
+		return nil, fmt.Errorf("could not find subnets for network %s", networkID)
+	}
+	return subs, nil
```

Comment:
gophercloud supports sorting via neutron API

Reply:
Sorting is not necessary, because a) any matching subnet can be chosen and b) when deleting LBs again, there will be always any number of subnets wuth free IPs